src/interpolators.jl

```diff
@@ -128,6 +133,85 @@ function (sim::ScatteredInterpolationMethod)(
 
 end
 
+"""
+    NaturalNeighboursMethod(; method=Sibson(1), kwargs...)
+
+Interpolator that uses the [NaturalNeighbours.jl](https://github.com/DanielVandH/NaturalNeighbours.jl) package
+to interpolate the data.  This uses Delaunay triangulations and 
+the corresponding Voronoi diagram to interpolate the data, and offers
+a variety of methods like `Sibson(::Int)`, `Nearest()`, and `Triangle()`.
+
+The advantage of Voronoi-diagram based methods is that they are more robust to 
+irregularly distributed datasets and some discontinuities, which may throw off 
+some polynomial based methods as well as independant distance weighting (kriging). 
+See [this Discourse post](https://discourse.julialang.org/t/ann-naturalneighbours-jl-natural-neighbour-interpolation-and-derivative-generation/99164/11) for more information on why NaturalNeighbours are cool! 
+
+To access the methods easily, you should run `using NearestNeighbours`.
+
+See the [NaturalNeighbours documentation](https://github.com/DanielVandH/NaturalNeighbours.jl) for more details.
+
+This method is fully configurable and will forward all arguments to the relevant NaturalNeighbours.jl functions.
+To understand what the methods do, see the documentation for [`NaturalNeighbours.jl`](https://github.com/DanielVandH/NaturalNeighbours.jl).
+
+## Keyword arguments
+
+The main keyword argument to look at here is `method`, which is the method
+to use for the interpolation.  The other keyword arguments are simply forwarded 
+to NaturalNeighbours.jl's interpolator.
+
+$(Makie.DocStringExtensions.FIELDS)
+"""
+@with_kw struct NaturalNeighboursMethod <: Interpolator
+    "The method to use for the interpolation.  Defaults to `Sibson(1)."
+    method::NaturalNeighbours.AbstractInterpolator = NaturalNeighbours.Sibson{1}()
```

Comment:
Just another comment while I remember: It would be better to use `Sibson(1)` instead of `Sibson{1}()` here, I think that's the API I provide.

I wonder if `Sibson()` would be a better default. You are always generating derivatives since `derivatives=true` below, so `Sibson(1)` will always work, but I think this might be a problematic default for non-differentiable data. Maybe most data used by `TopoPlots.jl` is first-order differentiable, in which case this is fine. (The default I use in NaturalNeighbours.jl is `Sibson()` for this reason.)

Reply:
I picked Sibson(1) because the interpolation seemed a bit smoother to me, but can definitely change back to `Sibson` if it's more generally applicable.



```suggestion
    method::NaturalNeighbours.AbstractInterpolator = NaturalNeighbours.Sibson()
```

Reply:
Yeah `Sibson(1)` is definitely a lot smoother - `Sibson()` can look a bit rough sometimes... I probably should've made the comment more as a note than as a suggestion. Maybe keeping `Sibson(1)` as a default is fine - I imagine most data put into this package is smooth enough anyway. Feel free to keep it as is.